src/models/tests/gamma_poisson_model_test.py

```diff
@@ -44,131 +44,136 @@ def gamma_poisson_pmf(k, alpha, beta):
-        gpm = GammaPoissonModel([ReachPoint([], [])])
-        alpha, beta, I = gpm._feasible_point([14, 8, 4, 2])
-        self.assertAlmostEqual(alpha, 1.0)
-        self.assertAlmostEqual(beta, 56 / 500)
-        self.assertAlmostEqual(I, 500)
+        self.assertAlmostEqual(h_actual[0], 6)
+        self.assertAlmostEqual(h_actual[1], 3 / 2)
+        self.assertAlmostEqual(h_actual[2], 3 / 8)
 
     def test_fit_histogram_chi2_distance(self):
-        gpm = GammaPoissonModel([ReachPoint([], [])])
-        I0, N0, alpha0, beta0 = 20000, 10000, 5, 0.5
-        h_actual = gpm._expected_histogram(I0, I0 / 0.1, N0, alpha0, beta0)
-        Imax, N, alpha, beta = gpm._fit_histogram_chi2_distance(h_actual)
-        Ih = np.sum([h_actual[i] * (i + 1) for i in range(len(h_actual))])
-        h_predicted = gpm._expected_histogram(Ih, Imax, N, alpha, beta)
+        # The following point corresponds to
+        #  I, Imax, N, alpha0, beta0 = 5000, 30000, 10000, 1.0, 2.0
+        # This represents a distribution with mean = 3 and variance = 6.
+        h_actual = [2812, 703, 175, 43, 10, 2, 0, 0, 0, 0]
+        kplus_actual = [3745, 933, 230, 55, 12, 2, 0, 0, 0, 0]
+        rp = ReachPoint([5000], kplus_actual)
+        gpm = GammaPoissonModel([rp])
+        Imax, N, alpha, beta = gpm._fit_histogram_chi2_distance(rp, nstarting_points=1)
+        h_predicted = gpm._expected_histogram(5000, Imax, N, alpha, beta)
         for i in range(len(h_actual)):
             self.assertTrue(
-                (h_actual[i] - h_predicted[i]) ** 2 < 1,
+                (h_actual[i] - h_predicted[i]) ** 2 / h_predicted[i] < 1,
                 f"Discrepancy found at position {i}. "
                 f"Got {h_predicted[i]} Expected {h_actual[i]}",
             )
 
     def test_fit_histogram_fixed_N(self):
-        gpm = GammaPoissonModel([ReachPoint([], [])])
-        I0, N0, alpha0, beta0 = 4000, 10000, 1, 1
-        h_actual = gpm._expected_histogram(I0, I0 / 0.1, N0, alpha0, beta0)
-        Imax, alpha, beta = gpm._fit_histogram_fixed_N(h_actual, N0)
-        Ih = np.sum([h_actual[i] * (i + 1) for i in range(len(h_actual))])
-        h_predicted = gpm._expected_histogram(Ih, Imax, N0, alpha, beta)
+        # The following point corresponds to
+        #  I, Imax, N, alpha0, beta0 = 15000, 30000, 10000, 1.0, 2.0
+        # This represents a distribution with mean = 3 and variance = 6.
+        h_actual = [3749, 1875, 937, 468, 234, 117, 58, 29, 14, 7]
+        kplus_actual = [7488, 3739, 1864, 927, 459, 225, 108, 50, 21, 7]
+        rp = ReachPoint([15000], kplus_actual)
+        gpm = GammaPoissonModel([rp])
+        Imax, alpha, beta = gpm._fit_histogram_fixed_N(rp, 10000)
+        h_predicted = gpm._expected_histogram(15000, Imax, 10000, alpha, beta)
         for i in range(len(h_actual)):
             self.assertTrue(
-                (h_actual[i] - h_predicted[i]) ** 2 < 1,
+                (h_actual[i] - h_predicted[i]) ** 2 / h_predicted[i] < 1,
```

Comment:
Now i = 0 (i.e., 1 reach), the discrepancy is 0.236.
For all i >= 1, (h_actual[i] - h_predicted[i]) ** 2 / h_predicted[i] < 0.1.

How is it compared to the performance of the previous optimizer?

Reply:
Here are the parameter values:

```
True values:      Imax = 30000, alpha = 1, beta = 2
New optimizer: Imax = 30454, alpha = 1.03, beta = 1.98
Old optimizer:  Imax = 148880, alpha = 15.47, beta = 1.04

Here are the frequency histograms:
True: [3749, 1875, 937, 468, 234, 117, 58, 29, 14, 7]
New: [3719, 1886, 945, 471, 234, 116, 57, 28, 14, 7]
Old:  [3041, 2521, 1464, 669, 256, 85, 25, 6, 1, 0]
```

Reply:
Impressive!  Leaving the comment open for other reviewers.